apps/spotrem/app.js

```diff
@@ -1,305 +1,375 @@
-        Bangle.removeListener("swipe", swipeHandler);
+        Bangle.removeListener('message', messageHandler);
+        Bangle.removeListener('audio', audioHandler);
+        if (volumeSlider) {
+          volumeSlider.f.remove();
+        }
+        if (progressBar) {
+          progressBar.f.stopAutoUpdate();
+          //progressBar.f.remove();
+        }
         clearWatch(buttonHandler);
         widgetUtils.show();
       }
-    },
-      ud => {
-        if (ud) Bangle.musicControl(ud>0 ? "volumedown" : "volumeup");
-      }
-  );
-  Bangle.on("touch", touchHandler);
-  Bangle.on("swipe", swipeHandler);
-  let buttonHandler = setWatch(()=>{load();}, BTN, {edge:'falling'});
-};
-
-// Get back to the main layout
-let backToGfx = function() {
-  E.showMenu();
-  g.clear();
-  g.reset();
-  setUI();
+    });
+    let buttonHandler = setWatch(()=>{load();}, BTN, {edge:'falling'});
+  };
+
+  // Get back to the main layout
+  let backToGfx = function() {
+    E.showMenu();
+    g.clear();
+    g.reset();
+    gfx();
+    setUI();
+    progressBar.f.autoUpdate();
+    progressBar.f.startAutoUpdate();
+    print("proglevel: ", progressBar.v.level)
+    progressBar.f.draw(progressBar.v.level);
+    backToMenu = false;
+  };
+
+  ////////
+  //The functions for interacting with Android and the Spotify app
+  ////////
+
+  let createCommand = function(o) {
+    return ()=>{
+      Bluetooth.println("");
+      Bluetooth.println(JSON.stringify(o));
+    };
+  };
+
+  let assembleSearchString = function() {
+    return (artist=="" ? "":("artist:\""+artist+"\"")) + ((artist!="" && track!="") ? " ":"") + (track=="" ? "":("track:\""+track+"\"")) + (((artist!="" && album!="") || (track!="" && album!="")) ? " ":"") + (album=="" ? "":(" album:\""+album+"\""));
+  };
+
+  simpleSearch = "";
+  let simpleSearchTerm = function() { // input a simple search term without tags, overrides search with tags (artist and track)
+    require("textinput").input({text:simpleSearch}).then(result => {simpleSearch = result;}).then(() => {E.showMenu(searchMenu);});
+  };
+
+  artist = "";
+  let artistSearchTerm = function() { // input artist to search for
+    require("textinput").input({text:artist}).then(result => {artist = result;}).then(() => {E.showMenu(searchMenu);});
+  };
+
+  track = "";
+  let trackSearchTerm = function() { // input track to search for
+    require("textinput").input({text:track}).then(result => {track = result;}).then(() => {E.showMenu(searchMenu);});
+  };
+
+  album = "";
+  let albumSearchTerm = function() { // input album to search for
+    require("textinput").input({text:album}).then(result => {album = result;}).then(() => {E.showMenu(searchMenu);});
+  };
+
+  let searchPlayWOTags = createCommand({t:"intent", action:"android.media.action.MEDIA_PLAY_FROM_SEARCH", categories:["android.intent.category.DEFAULT"], package:"com.spotify.music", target:"activity", extra:{query:simpleSearch}, flags:["FLAG_ACTIVITY_NEW_TASK"]});
```

Comment:
Thanks! I guess it could be taken further. There's pretty much duplication inside the command objects (e.g. `categories: ["android.intent.category.DEFAULT"]`). But this change was pretty easy to start with.